cleanup.sh

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bash
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eu
+
+# Should this script error, print out the line that was responsible
+_error_report() {
+  echo >&2 "Exited [$?] at line $(caller):"
+  cat -n $0 | tail -n+$(($1 - 3)) | head -n7 | sed "4s/^\s*/>>> /"
+}
+trap '_error_report $LINENO' ERR
+
+
+export REGION=${REGION:=us-central1}
+
+if [[ -z "${PROD_PROJECT}" ]]; then
+    echo "Please set the $(tput bold)PROD_PROJECT$(tput sgr0) variable"
+    exit 1
+elif [[ -z "${STAGE_PROJECT}" ]]; then
+    echo "Please set the $(tput bold)STAGE_PROJECT$(tput sgr0) variable"
+    exit 1
+elif [[ -z "${OPS_PROJECT}" ]]; then
+    echo "Please set the $(tput bold)OPS_PROJECT$(tput sgr0) variable"
+    exit 1
+fi
+
+# TF Resources
+STATE_GCS_BUCKET_NAME="$OPS_PROJECT-tf-states"
+
+PROD_ENVIRONMENT_DIR=terraform/environments/prod
+terraform -chdir=${PROD_ENVIRONMENT_DIR} destroy --auto-approve && rm ${PROD_ENVIRONMENT_DIR}/terraform.tfvars && rm -rf ${PROD_ENVIRONMENT_DIR}/.terraform
+
+STAGE_ENVIRONMENT_DIR=terraform/environments/staging
+terraform -chdir=${STAGE_ENVIRONMENT_DIR} destroy --auto-approve && rm ${STAGE_ENVIRONMENT_DIR}/terraform.tfvars && rm -rf ${STAGE_ENVIRONMENT_DIR}/.terraform
+
+OPS_ENVIRONMENT_DIR=terraform/environments/ops
```

Comment:
suggestion: You previously mentioned that if this project is deleted before the other environments, there can be trouble getting terraform destroy to work. (or maybe it was specifically the bucket below). It would be great to get a comment on that to help us avoid accidentally breaking this script in the future.

Reply:
I'll add a comment indicating that ops resources should be deleted last.